sdk/test-utils/recorder/src/utils.ts

```diff
@@ -21,17 +21,73 @@ export function isPlaybackMode() {
   return !isRecordMode() && !isLiveMode();
 }
 
+/**
+ * Encodes a string as a URI component, but also taking in consideration the RFC 3986 specification.
+ * JavaScript's encodeURIComponent method doesn't take in consideration the characters: !, ', (, ), *
+ * @param str The string that needs to be encoded.
+ */
+export function encodeRFC3986(str: string): string {
+  return encodeURIComponent(str).replace(
+    /[!'()*]/g,
+    (x) =>
+      `%${x
+        .charCodeAt(0)
+        .toString(16)
+        .toUpperCase()}`
+  );
+}
+
+/**
+ * Escapes all of the valid RegExp characters of a string.
+ * @param str The string that needs to be escaped.
+ */
 export function escapeRegExp(str: string): string {
-  return encodeURIComponent(str)
-    .replace(
-      /[!'()*]/g,
-      (x) =>
-        `%${x
-          .charCodeAt(0)
-          .toString(16)
-          .toUpperCase()}`
-    ) // RFC 3986.
-    .replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&"); // All the RegExp sensitive characters.
+  return str.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&");
+}
+
+export type ReplacementDictionary = { [x: string]: string };
+
+/**
+ * Looks for the environment variables based on the keys of the given dictionary,
+ * then replaces the values found with each value from the same dictionary.
+ * @param replacements A dictionary of string keys and string values.
+ * @param content The content that has the text to be replaced.
+ */
+export function applyReplacementDictionary(
+  env: { [x: string]: string },
+  replacements: ReplacementDictionary,
+  content: string
+): string {
+  let updated = content;
+  for (const k of Object.keys(replacements)) {
+    if (env[k]) {
+      updated = updated.replace(
+        new RegExp(escapeRegExp(encodeRFC3986(env[k])), "g"),
```

Comment:
minor suggestion: do you think it would be worth it to extract a utility called `replaceAll` that handles calling `replace` on the string?

Reply:
then it would just be `updated = replaceAll(updated, encodeRFC3986(env[k]), encodeRFC3986(replacements[k]));`

Reply:
Careful with replaceAll that behaves differently from [this proposal](https://github.com/tc39/proposal-string-replaceall) (which can't come soon enough!!)

Reply:
Here's my current replaceAll:

```ts
function replaceAll(string: string, pattern: string, replacement: string) {
  return string.replace(new RegExp(escapeRegExp(pattern), "g"), replacement);
}
```